x-pack/plugins/ingest_manager/public/applications/ingest_manager/hooks/use_core.ts

```diff
@@ -4,15 +4,13 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import React, { useContext } from 'react';
-import { CoreStart } from 'src/core/public';
+import { CoreStart } from 'kibana/public';
+import { useKibana } from '../../../../../../../src/plugins/kibana_react/public';
 
-export const CoreContext = React.createContext<CoreStart | null>(null);
-
-export function useCore() {
-  const core = useContext(CoreContext);
-  if (core === null) {
-    throw new Error('CoreContext not initialized');
+export function useCore(): CoreStart {
+  const { services } = useKibana();
+  if (services === null) {
+    throw new Error('KibanaContextProvider not initialized');
   }
-  return core;
+  return services;
```

Comment:
will return the same thing that core returned. could be renamed to say core if that clears up any confusion.

Reply:
I don't mind keeping it as `services` here for now as it's largely an abstraction anyway. On our team's side we can consider renaming `useCore` as tech debt

Reply:
@jen-huang FYI - Kibana Arch. also provide a hook that is very similar to Ingest's `useCore()` - `useKibana()` is the hook that goes along with`KibaanContextProvider` and that is what we are using in SIEM/endpoint